lib/DidJwtStrategy.js

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const async = require('async');
+const bedrock = require('bedrock');
+const config = bedrock.config;
+const jwt = require('jsonwebtoken');
+let jsonld = bedrock.jsonld;
+let	jsigs = require('jsonld-signatures');
+const passport = require('passport');
+const BedrockError = bedrock.util.BedrockError;
+require('bedrock-passport');
+
+let didio;
+bedrock.events.on('bedrock.ready', () => {
+  // create local copy of jsonld
+  jsonld = jsonld();
+  // set didio defaults
+  didio = require('did-io')(config.passport.strategies.did.didio);
+  // this allows custom document loaders to be configured
+  didio.use('jsonld', jsonld);
+  // use didio when getting JSON-LD docs and wrap the bedrock.jsonld
+  // document loader
+  jsonld.documentLoader = didio.createDocumentLoader({
+    wrap: function(url, callback) {
+      return bedrock.jsonld.documentLoader(url, callback);
+    }
+  });
+  // create local jsigs copy and use local jsonld copy
+  jsigs = jsigs();
+  jsigs.use('jsonld', jsonld);
+});
+
+/**
+ * Creates a new didJwt Strategy for use with passport.
+ */
+class Strategy extends passport.Strategy {
+  constructor(options) {
+    super();
+    this.name = 'didJwt',
+    this.options = options;
+    this.store = null;
+  }
+}
+
+module.exports = Strategy;
+
+/**
+ * Authenticate a request.
+ *
+ * @param req the request to authenticate.
+ */
+Strategy.prototype.authenticate = function(req) {
+  const self = this;
+  if(!(`authorization` in req.headers)) {
+    return self.error(new BedrockError(
+      '`Authorization` header is required.', 'InvalidHeader'));
+  }
+  const authHeader = req.headers.authorization.split(' ');
+  if(authHeader.length !== 2 || authHeader[0].toLowerCase() !== 'bearer') {
+    return self.error(new BedrockError(
+      '`Authorization` header format must be `Bearer <token>`.',
+      'InvalidHeader', {
+        header: req.headers.authorization
+      }));
+  }
+  // FIXME: it would be nice to move some of this to verify as well
+  const token = authHeader[1];
+
+  self.verify(token, (err, payload) => {
+    if(err) {
+      return self.error(new BedrockError(
+        'Invalid Token.', 'InvalidToken', {}, err));
+    }
+    req.user = {
+      identity: {id: payload.sub}
+    };
+    self.success(req.user);
+  });
+};
+
+Strategy.prototype.init = function(callback) {
+  const self = this;
+  if(self.store) {
+    async.parallel([
+      callback => async.forEachOf(
+        config['authn-did-jwt'].crossDomainAuthn.namespaces, (v, k, callback) =>
+          self.store.provision(v, callback), callback),
```

Comment:
This was a bit of a mind bender due to the current style...

Maybe change this:

```js
callback => async.forEachOf(
  config['authn-did-jwt'].crossDomainAuthn.namespaces, (v, k, callback) =>
    self.store.provision(v, callback), callback),
callback => async.forEachOf(...
```

To this?

```js
callback => async.forEachOf(
  config['authn-did-jwt'].crossDomainAuthn.namespaces,
  (v, k, callback) =>  self.store.provision(v, callback),
  callback),
callback => async.forEachOf(...
```